### ISSUE TITLE
Feature Request: do not apply certain policies to selected users/groups/service accounts

### ISSUE THREAD

**Issue by flavio** [kubewarden/kubewarden-controller#554]
### Is your feature request related to a problem?

This is a feature request, based on a conversation happened on slack.

> Hi! that's an interesting question.
The user name, uid, and groups are present on the userInfo  key of the admissionReviews that come from k8s apiserver, so it's possible, but AFAIK right now we don't have any policy written by the Kubewarden team that makes use of it.
It could be doable by writing a policy, running Rego policy or a Kyverno one (experimental still) in Kubewarden. (edited) 
>
> At the moment we as team plans to have the following fields to be part of settings of every policy
policy_not_applied_to_groups, policy_not_applied_to_service_accounts
Both of these are arrays. In our implementation, we compare these with the validationRequest.Request.UserInfo.Groups and validationRequest.Request.UserInfo.Username. If it matches we simply do kubewarden.AcceptRequest(). Else, we do the other validation and accepts requests only if the validations pass.
I want to know if there are other ways to do this without mentioning the groups or ids in the settings (edited)

**Comment by anishnagaraj** [comment]
I like the 3rd option in terms of the control that the admins get. But in my view this sensitive data. So I would sugggest a way where this data is not part of any of the git commits. The key of this data can be used as part of the policy but the value itself should be configured inside the server, where data is pulled from a external secret.

**Comment by flavio** [comment]
Yes, the implementation of the policy that checks for the trusted users/groups/service accounts is entirely up to the policy author. We could of course provide a reference implementation.

There could be different ways to feed this data (the allow list) into the policy. Doing that by hard coding the values inside of the policy would be the worse way.
A simpler approach would be to leverage the policy settings. Basically this data would be read specified inside of the `ClusterAdmissionPolicy` / `AdmissionPolicy` yaml.
Another option would be to make this policy context aware and read the allow list from a specific Configmap or Secret.

I would personally prefer the settings route because:
* Policy definition + configuration stays in the same place. When using a Secret/Configmap there's a layer of indirection
* I don't think this data is confidential, hence worth of a Secret. Keep in mind that you can remove read access to the `ClusterAdmissionPolicy`/`AdmissionPolicy` resources via RBAC. Users won't be able to see their configuration options if you want to
* Context-aware policies hit the Kubernetes API. Sure, we do some caching, but this could increase the load on the API server if your policy is invoked often. In this case, given the previous points. I don't think this is worth.

Again, this is my personal taste. You could write your own policy that works in a different way :)

**Comment by anishnagaraj** [comment]
I would go for option 3#
